packages/salesforce/src/Adaptor.js

```diff
@@ -646,69 +646,90 @@ export function reference(position) {
   return state => state.references[position].id;
 }
 
-function setApiVersion(apiVersion) {
+function setApiVersion(state, connectionOptions) {
+  const { apiVersion } = state.configuration;
+
   const apiVersionRegex = /^\d{2}\.\d$/;
-  let version = '52.0';
+
   if (apiVersion && apiVersionRegex.test(apiVersion)) {
     console.log('Using Salesforce API version', apiVersion);
-    version = apiVersion;
+    return { ...connectionOptions, version: apiVersion };
   } else {
-    console.log('Invalid salesforce apiVersion', apiVersion);
-    console.log('Using Salesforce API version', version);
+    console.log('apiVersion is not defined');
+    console.log('We recommend using Salesforce API version 52.0 or latest');
+    return connectionOptions;
   }
+}
+
+/**
+ * Establish a connection to Salesforce using username and password.
+ * @function createBasicAuthConnection
+ * @private
+ * @param {State} state - Runtime state.
+ * @returns {State}
+ */
+async function createBasicAuthConnection(state) {
+  const { loginUrl, username, password, securityToken } = state.configuration;
 
-  return version;
+  const connectionOptions = setApiVersion(state, { loginUrl });
+
+  const connection = new jsforce.Connection(connectionOptions);
+
+  await connection
+    .login(username, securityToken ? password + securityToken : password)
+    .catch(e => {
+      console.error(`Failed to connect to salesforce as ${username}`);
+      throw e;
+    });
+
+  console.info(`Connected to salesforce as ${username}.`);
+
+  return {
+    ...state,
+    connection,
+  };
 }
+
 /**
- * Creates a connection.
- * @example
- * createConnection(state)
- * @function
+ * Establish a connection to Salesforce using Access Token.
+ * @function createAccessTokenConnection
+ * @private
  * @param {State} state - Runtime state.
  * @returns {State}
  */
-function createConnection(state) {
-  const { loginUrl, apiVersion } = state.configuration;
 
-  if (!loginUrl) {
-    throw new Error('loginUrl missing from configuration.');
-  }
+function createAccessTokenConnection(state) {
+  const { other_params, access_token } = state.configuration;
+  const { instance_url } = other_params;
+
+  const connectionOptions = setApiVersion(state, {
+    instanceUrl: instance_url,
+    accessToken: access_token,
+  });
+
+  const connection = new jsforce.Connection(connectionOptions);
```

Comment:
Do we not need to do any sort of `login` here?

The connection is synchronous, how do we know we're actually connected?

Reply:
No login there, It's a different strategy based on auth type, 
So for basic-auth you will setup the connection then do `connection.login(connectionOptions)`. And for accessToken auth, `getConnection(state, connOpts)` will have the connection with session type `oauth`

For Basic Auth - https://jsforce.github.io/document/#username-and-password-login
For OAuth - https://jsforce.github.io/document/#access-token